src/playback-watcher.js

```diff
@@ -211,6 +211,19 @@ export default class PlaybackWatcher {
       this.timer_ = setTimeout(this.skipTheGap_.bind(this),
                                difference * 1000,
                                currentTime);
+      return;
+    }
+
+    // Sometimes the player can stall for unknown reasons within a contiguious buffered
+    // region with no indication that anything is amiss (Seen in Firefox). Seeking to
+    // currentTime is usually enough to kickstart the player
+    if (buffered.length && currentTime + 1 < buffered.end(buffered.length - 1)) {
```

Comment:
With really short segments it's possible this can break. We may want to use a function that checks to see if we're within a buffered region (and if we want to be careful around edges, we can use a margin of some partial second, as 1 may be too long).